compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/JvmCachedDeclarations.kt

```diff
@@ -182,6 +186,29 @@ class JvmCachedDeclarations(
             }
         }
 
+    fun getStaticCompanionReplacementDeclaration(jvmStaticFunction: IrSimpleFunction): IrSimpleFunction =
+        staticCompanionReplacementDeclarations.getOrPut(jvmStaticFunction) {
+            val companion = jvmStaticFunction.parentAsClass
+            assert(companion.isCompanion)
```

Comment:
Please, add lambda to `assert` call, otherwise, there will be unhelpful `assertion failed` without further explanation.

Reply:
Apologies. This was done this way because this is what the method above it (which gets the static proxy from cache) does. I'll add this.